javagen/src/main/java/com/azure/autorest/mapper/ModelMapper.java

```diff
@@ -244,6 +246,9 @@ public ClientModel map(ObjectSchema compositeType) {
             List<ClientModelPropertyReference> propertyReferences = new ArrayList<>();
             for (Property property : compositeTypeProperties) {
                 ClientModelProperty modelProperty = Mappers.getModelPropertyMapper().map(property);
+                if (settings.isOutputImmutable() && !compositeType.getUsage().contains(SchemaContext.INPUT)) {
+                    modelProperty = modelProperty.newBuilder().isReadOnly(true).build();
+                }
```

Comment:
I cannot think of a better solution.

We want to set all Properties to read-only, if Model is not used as input. But the `ModelPropertyMapper` only sees the Property, not the Model.

Doing it in template is too hard (tried and aborted).